app/main.py

```diff
@@ -173,57 +193,95 @@ def generate_prometheus_metrics(metrics):
-                                                                            datetime.fromtimestamp(resp['datapoints'][0]['timestamp'] / 1000.0),
-                                                                            resp['datapoints'][0]['average']))
-
-        elif r.status_code == 401:
-            logging.warn("Token seems to be expired, requesting a new one and retrying")
-            request_token()
-            get_metric_value(prometheus_metrics=prometheus_metrics, metrics=metrics)
-            break
-        else:
-            logging.error("{0} for '{1}={2}' at {3} got HTTP Status Code: {4}".format(full_metric_name, dimensions_name,
-                                                                                      dimensions_value, current_time,
-                                                                                      r.status_code))
+    metrics_set = []
+
+    # It seems that we cannot batch metrics from different Namespaces
+    # Therefore we iterate over the wanted Namespaces
+    wanted_namespaces = ['SYS.%s' % n for n in config.get('EXPORTER_CONFIG', 'namespaces').split(',')]
+    for wn in wanted_namespaces:
+        # filter metrics
+        filtered = [ item for item in metrics if item["namespace"] == wn ]
+
+        # For each OTC metric from the filtered Namespace
+        for idx, m in enumerate(filtered, start=1):
+            namespace = m["namespace"]
+            metric_name = m["metric_name"]
+            dimensions_name = m["dimensions"][0]["name"]
+            dimensions_value = m["dimensions"][0]["value"]
+            full_metric_name = "%s:%s" % (namespace, metric_name)
+
+            # For ELB skip metrics per listener, as I don't know how to represent them yet and they would overwrite the overall ELB metric
+            if namespace == "SYS.ELB" and len(m["dimensions"]) > 1:
+                continue
+
+            metric = {
+                "namespace": namespace,
+                "dimensions": [
+                    {
+                        "name": dimensions_name,
+                        "value": dimensions_value
+                    }
+                ],
+                "metric_name": metric_name
+            }
+
+            metrics_set.append(metric)
+
+            # After each 10 metrics query them as a batch
+            if idx % 10 == 0 or idx == len(filtered):
+
+                # Create payload for batch request
+                payload = {
+                    "metrics": metrics_set,
+                    "from": current_time[0],
+                    "to": current_time[1],
+                    "period": "300",
+                    "filter": "average"
+                }
+
+                # Reset metrics_set for next batch
+                metrics_set = []
+
+
+                # Wait for 100 milliseconds before every request (needed to don't overload OTC API)
+                time.sleep(0.1)
```

Comment:
I know the previous version didn't have it, but what do you think about adding a exponential backoff retry strategy?

Reply:
Sounds like a good idea. Will try to investigate.

Reply:
> Can you please check if it makes sense to add a exponential backoff retry strategy?

@tiagoReichert 
That sounds promising: https://stackoverflow.com/a/35504626
What do you think?

Reply:
Yes, I think it makes sense. Can you add this to the PR?